site-builder/src/site/resource.rs

```diff
@@ -399,6 +399,35 @@ impl ResourceManager {
         )))
     }
 
+    ///  Derives the HTTP headers for a resource based on the ws-resources.yaml.
+    ///
+    ///  Matches the path of the resource to the wildcard paths in the configuration to
+    ///  determine the headers to be added to the HTTP response.
+    fn derive_http_headers(&self, resource_path: &str) -> BTreeMap<String, String> {
+        self.ws_resources
+            .as_ref()
+            .and_then(|config| config.headers.as_ref())
+            .and_then(|headers| {
+                headers
+                    .iter()
+                    .filter(|(path, _)| ResourceManager::is_pattern_match(path, resource_path))
+                    .max_by_key(|(path, _)| path.split('/').count()) // Longest path match
+                    .map(|(_, header_map)| header_map.0.clone())
+            })
+            .unwrap_or_default()
+    }
+
+    /// Matches a pattern to a resource path.
+    ///
+    /// The pattern can contain a wildcard `*` which matches any sequence of characters.
+    /// e.g. `foo/*` will match `foo/bar` and `foo/bar/baz`.
```

Comment:
Nit: a resource path has a leading `/`.
```suggestion
    /// e.g. `/foo/*` will match `/foo/bar` and `/foo/bar/baz`.
```